### ISSUE TITLE
Make `check-types` configurable

### ISSUE THREAD

**Issue by golopot** [gajus/eslint-plugin-jsdoc#215]
Users should be able to set the preference of types. The option could looks like this:
```js
{
  '@typescript-eslint/ban-types': [
    'error',
    {
      types: {
        // ban type `Foo` using the default error message
        Foo: null,

        // ban type `Bar` with a custom message.
        Bar: "Don't use Bar!",

        // ban type `String` and tell the plugin how to fix it.
        String: {
          message: 'Use string instead',
          fixWith: 'string',
        },
      },
    },
  ],
};
```

See [@typescript-eslint/ban-types](https://github.com/typescript-eslint/typescript-eslint/blob/master/packages/eslint-plugin/docs/rules/ban-types.md).

As a side note, `ban-types` is a better name than `check-types`. The name `check-types` could mean "check type is valid", "check type is consistent to how it is used" or "check the style of types".

**Comment by brettz9** [comment]
See my [comment](https://github.com/gajus/eslint-plugin-jsdoc/issues/166#issuecomment-480778864) on a related issue, i.e., per issue #107 I think we might look at something like this if reimplementing valid-jsdoc's `preferType`.

**Comment by brettz9** [comment]
Closed by #247. Thanks for the great idea! I tweaked it a little in the implementation, but it is similar as you can see in the docs.
